fetch_data.py

```diff
@@ -25,27 +30,41 @@
 import random
 from docopt import docopt
 from pathlib import Path
+from enum import Enum
 
 
 TWITTER_RATE_LIMIT_ERROR = 88
 
 
-def fetch_users(apis, target, are_users, nodes_to_consider, max_tweets_count, out_path,
+class Mode(Enum):
+    USERS = 1
+    TWEETS = 2
+    LIKES = 3
+
+
+def fetch_users(apis, target, mode, nodes_to_consider, max_tweets_count, out_path,
                 followers_file="cache/followers.json",
                 friends_file="cache/friends.json",
                 tweets_file="cache/tweets.json"):
     """
         Fetch a list of users from Twitter API.
 
-        - If a target (user) is provided, get their friends and followers.
-        - Alternatively, if a search query is provided, get the resulting tweets and their users.
-          These users are returned as "followers" of the query, and the list of friends is None.
+        - If in USERS mode, the target is a username. We retrieve and cache the user's friends and followers.
+        - If in TWEETS mode, the target is a search query. We retrieve and cache the tweets associated with the query,
```

Comment:
Remark: is is not great that the authors of tweets (either from a search query or likes) are referred to `followers` (and empty `friends`) rather than `authors`. But it was already the case and should be fixed independently of this PR, so it's good that you kept the same structure.

Reply:
Acknowledged.